src/librustc/ty/item_path.rs

```diff
@@ -109,16 +128,32 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
                         ..
                     }) = *opt_extern_crate
                     {
+                        debug!("push_krate_path: def_id={:?}", def_id);
                         self.push_item_path(buffer, def_id);
                     } else {
-                        buffer.push(&self.crate_name(cnum).as_str());
+                        let name = self.crate_name(cnum).as_str();
+                        debug!("push_krate_path: name={:?}", name);
+                        buffer.push(&name);
                     }
+                } else if self.sess.edition() == Edition::Edition2018 {
+                    SHOULD_PREFIX_WITH_CRATE.with(|flag| {
```

Comment:
Ugh. Totally pre-existing, but this code is so wickedly complex.

/me feels responsible

Reply:
The reason I added a `thread_local!` variable for this was so that we didn't add the prefix when generating an item path for other types of diagnostics that weren't suggesting that you `use` a item path. If there's a preferred alternative approach, I'm happy to change this.